subsys/net/lib/lwm2m/lwm2m_obj_device.c

```diff
@@ -219,6 +252,51 @@ int lwm2m_update_device_service_period(uint32_t period_ms)
 	return lwm2m_engine_update_service_period(device_periodic_service, period_ms);
 }
 
+#ifdef CONFIG_LWM2M_DEVICE_ERROR_CODE_SETTINGS
+static int lwm2m_obj_device_settings_set(const char *name, size_t len,
+					 settings_read_cb read_cb, void *cb_arg)
+{
+	const char *next;
+	int rc;
+	int i;
+
+	if (settings_name_steq(name, ERROR_LIST_KEY, &next) && !next) {
+		if (len > sizeof(error_code_list)) {
+			LOG_ERR("Error code list too large: %u", len);
+			return -EINVAL;
+		}
+
+		rc = read_cb(cb_arg, error_code_list, sizeof(error_code_list));
+		if (rc == 0) {
+			reset_error_list();
+			return 0;
+		} else if (rc > 0) {
+			for (i = 0; i < sizeof(error_code_list); i++) {
```

Comment:
```suggestion
			for (i = 0; i < ARRAY_SIZE(error_code_list); i++) {
```

If someone changes the storage type of error_code_list to something else than `uint8_t` the for-loop would have caused buffer overflow.
`ARRAY_SIZE()` is safer.

Reply:
Done